### PR TITLE
Code cleanups and WinRM setup

### PR DESCRIPTION
**Pull request by adjackura**
instance_setup.ps1
  - Add WinRM setup
  - Code and style cleanups

gce_base.psm1
  - Cleanup deprecated functions
  - Deprecate some additional functions

### CODE REVIEW

File: gce/sysprep/instance_setup.ps1 [hunk 1]
```diff
@@ -167,7 +160,7 @@ function Change-InstanceName {
   }
 
   $count = 1
-  do {
+  while ($hostname_parts.Length -le 1) {
```
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: Where is hostname_parts getting set?

File: gce/sysprep/instance_setup.ps1 [hunk 2]
```diff
@@ -286,9 +243,12 @@ function Enable-RemoteDesktop {
       services.
```
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: If you are removing the spaces in between function docs in other files, you should do it everywhere.

File: gce/sysprep/instance_setup.ps1 [hunk 3]
```diff
@@ -314,6 +274,36 @@ function Enable-RemoteDesktop {
 }
 
 
+function Configure-WinRM {
```
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: I find the commands fairly confusing to read; can you make parameters more clear and add more comments?

File: gce/sysprep/instance_setup.ps1 [hunk 4]
```diff
@@ -504,7 +489,7 @@ $PSHome\powershell.exe -NoProfile -NoLogo -ExecutionPolicy Unrestricted -File "$
 
   try {
     # Call startup script during sysprep specialize phase.
-    _RunExternalCMD $script:metadata_script_loc 'specialize'
+    _RunExternalCMD $script:metadata_script_loc 'specialize'
```
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: Any idea why GitHub is saying this line changed?

File: gce/sysprep/instance_setup.ps1 (hunk 1 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: Below, its not getting set before this, so length is less than or equal to 1

File: gce/sysprep/instance_setup.ps1 (hunk 2 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: I'm not, or it wasn't my plan to. Where do you see me removing spaces?

File: gce/sysprep/instance_setup.ps1 (hunk 3 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: Added comments explaining usage

File: gce/sysprep/instance_setup.ps1 (hunk 2 above)
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: gce_base.psm1

File: gce/sysprep/instance_setup.ps1 (hunk 1 above)
Comment (illfelder) [inline gce/sysprep/instance_setup.ps1]: I think it's more confusing to use the property of a null object (which would throw an exception in every other language). The do while seems clearer.

File: gce/sysprep/instance_setup.ps1 (hunk 2 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: Aha, not sure why I missed that. Changed the rest to remove extra line as it makes long functions very long

File: gce/sysprep/instance_setup.ps1 (hunk 4 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: Yes, it was some odd character, fixed

File: gce/sysprep/instance_setup.ps1 (hunk 1 above)
Comment (adjackura) [inline gce/sysprep/instance_setup.ps1]: True, reverted

**Comment by illfelder** [comment]
LGTM for development.

**Pull request merged**